coreTable/CoreTable.ts

```diff
@@ -1061,6 +1061,34 @@ export class CoreTable<
         )
     }
 
+    duplicateColumn(slug: ColumnSlug, overrides: COL_DEF_TYPE) {
+        if (slug === overrides.slug) {
+            throw new Error(
+                `Cannot duplicate column: column slug '${slug}' matches the existing column`
+            )
+        }
+
+        const existingColumn = this.get(slug)
+        if (!existingColumn) {
+            throw new Error(`Cannot duplicate column: '${slug}' does not exist`)
+        }
+
+        return this.transform(
+            {
```

Comment:
I like it. This is maybe the pattern to use for the other transforms (just building a new column store then passing that to transform)